### ISSUE TITLE
Minor incident where lots of user pods died

### ISSUE THREAD

**Issue by minrk** [jupyterhub/mybinder.org-deploy#493]
I'm not sure this warrants an incident report, but writing it down here so it goes somewhere.

<img width="831" alt="screen shot 2018-02-27 at 15 09 20" src="https://user-images.githubusercontent.com/151929/36734043-a99cf3ca-1bd2-11e8-87d1-446e4ad99916.png">

at ~08:18 CET today (Feb 27) one of our nodes (gke-prod-a-ssd-pool-32-134a959a-k7f8) seemed to have a bit of a crisis. Looking at [its logs leading up to the event](https://console.cloud.google.com/logs/viewer?project=binder-prod&minLogLevel=0&expandAll=false&timestamp=2018-02-27T07:18:02.000000000Z&dateRangeStart=2018-02-27T06:27:00.280Z&interval=JUMP_TO_TIME&resource=gce_instance%2Finstance_id%2F210694033853395397&dateRangeEnd=2018-02-27T08:27:00.280Z), there are lots of intermittent errors, most frequently:

> 07:17:44.339820 26798 fs.go:418] Stat fs failed. Error: no such file or directory

Around 08:18, there is a flood of errors that largely look like failures to talk to docker. Perhaps most significant is docker itself exiting:

> 2018-02-27 07:18:50.000 UTC docker daemon exited

By around 08:26, it appears that kubernetes restarted everything on the node, coming back as an apparently new node. User pods on the node did not come back (I think this is expected, as user pods are never restarted), but everything else did. After this event, the node appeared in good health and has not had noticeable issues.

Regardless of the apparent lack of follow-up problems on the node, the node has been cordoned at ~15:00 CET and can be drained in an hour or two, in case the issue was systemic.

I'm not sure there are any action items. I think this was just a node having trouble, or perhaps kubernetes recovering as best it can from the docker daemon itself crashing. It did a pretty good job!

**Comment by choldgraf** [comment]
I wonder if it'd be worth having a specific section of the SRE guide for node failures of some kind or another (e.g. a situation where our stuff was working fine, but the node just went bonkers). I still don't have a good feeling for how to "know" that a node is failing, and having a place to briefly overview the kinds of errors etc around a node's failure might be helpful. What do you think?

**Comment by minrk** [comment]
yeah, a record of all the ways that nodes have failed (or been assumed to have failed) in the past, and how that conclusion was come to would be useful.

The graph above is the first one that's been really clear to me. Usually I look through the VM logs for "lots of errors" when I have a vague sense that "things are bad" and see if there's a prevalence on one node more than others.

**Comment by willingc** [comment]
I'll take the action item to write an abbreviated incident report so that we have one way for now to capture the incidents.

**Comment by choldgraf** [comment]
@willingc did we get an incident report merged? I don't see anything obviously related to this issue in there (maybe http://mybinder-sre.readthedocs.io/en/latest/incident-reports/2018-02-22-nginx-down.html ?)

alternatively, if this is minor I'm fine just closing this. What do you all think?
